src/main/java/org/takes/rs/RsPrint.java

```diff
@@ -115,41 +115,52 @@ public void print(final OutputStream output) throws IOException {
+     * Print it into output stream in UTF8.
      * @param output Output to print into
      * @throws IOException If fails
      * @since 0.10
      */
     public void printHead(final OutputStream output) throws IOException {
+        this.printHead(new Utf8OutputStreamWriter(output));
+    }
+
+    /**
+     * Print it into a writer.
+     * @param writer Writer to print into
+     * @throws IOException If fails
+     * @since 2.0
+     */
+    public void printHead(final Writer writer) throws IOException {
         final String eol = "\r\n";
-        final Writer writer =
-            new Utf8OutputStreamWriter(output);
         int pos = 0;
-        for (final String line : this.head()) {
-            if (pos == 0 && !RsPrint.FIRST.matcher(line).matches()) {
-                throw new IllegalArgumentException(
-                    String.format(
-                        // @checkstyle LineLength (1 line)
-                        "first line of HTTP response \"%s\" doesn't match \"%s\" regular expression, but it should, according to RFC 7230",
-                        line, RsPrint.FIRST
-                    )
-                );
-            }
-            if (pos > 0 && !RsPrint.OTHERS.matcher(line).matches()) {
-                throw new IllegalArgumentException(
-                    String.format(
-                        // @checkstyle LineLength (1 line)
-                        "header line #%d of HTTP response \"%s\" doesn't match \"%s\" regular expression, but it should, according to RFC 7230",
-                        pos + 1, line, RsPrint.OTHERS
-                    )
-                );
+        try {
+            for (final String line : this.head()) {
+                if (pos == 0 && !RsPrint.FIRST.matcher(line).matches()) {
+                    throw new IllegalArgumentException(
+                        String.format(
+                            // @checkstyle LineLength (1 line)
+                            "first line of HTTP response \"%s\" doesn't match \"%s\" regular expression, but it should, according to RFC 7230",
+                            line, RsPrint.FIRST
+                        )
+                    );
+                }
+                if (pos > 0 && !RsPrint.OTHERS.matcher(line).matches()) {
+                    throw new IllegalArgumentException(
+                        String.format(
+                            // @checkstyle LineLength (1 line)
+                            "header line #%d of HTTP response \"%s\" doesn't match \"%s\" regular expression, but it should, according to RFC 7230",
+                            pos + 1, line, RsPrint.OTHERS
+                        )
+                    );
+                }
+                writer.append(line);
+                writer.append(eol);
+                ++pos;
             }
-            writer.append(line);
             writer.append(eol);
-            ++pos;
+        } finally {
+            writer.flush();
```

Comment:
@victornoel the ticket #812 says:
> Implementation could not rely 
>  on simple try-finally block as OutputStreamWriter do not call 
>  to the flush() of OutputStream. It has additional checks. 

so this fix goes against requested changes.

Reply:
@g4s8 no it does not, as I explained in the PR description, the problem was that the test didn't work because `write` (which throws an exception for the sake of the test) was called during the `finally` phase of the try-finally block instead of inside the `try`.
I solved that situation (which was caused only by the test, not when using `RsPrint` for real) so now it is not the case anymore and it is working as expected.